exist-core/src/main/java/org/exist/xquery/Cardinality.java

```diff
@@ -1,91 +1,184 @@
- *  wolfgang@exist-db.org
- *  http://exist.sourceforge.net
- *  
- *  This program is free software; you can redistribute it and/or
- *  modify it under the terms of the GNU Lesser General Public License
- *  as published by the Free Software Foundation; either version 2
- *  of the License, or (at your option) any later version.
- *  
- *  This program is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU Lesser General Public License for more details.
- *  
- *  You should have received a copy of the GNU Lesser General Public License
- *  along with this program; if not, write to the Free Software
- *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
- *  
- *  $Id$
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 package org.exist.xquery;
 
+import static org.exist.xquery.Cardinality.InternalValue.*;
+
 /**
- * Defines occurrence indicators (*,?,+).
- * 
- * @author wolf
+ * Defines <a href="https://www.w3.org/TR/xpath-31/#prod-xpath31-OccurrenceIndicator">XPath Occurrence Indicators</a>
+ * (*,?,+), and additionally defines {@link #EMPTY_SEQUENCE}, and {@link #EXACTLY_ONE} for convenience.
+ *
+ * @author <a href="mailto:adam@evolvedbinary.com">Adam Retter</a>
  */
-public class Cardinality {
-
-	public final static int ZERO = 1;	
-	public final static int ONE = 2;	
-	public final static int MANY = 4;	
-	public final static int EMPTY = ZERO;
-	public final static int EXACTLY_ONE = ONE;
-	/** indicator '+' **/
-	public final static int ONE_OR_MORE = ONE | MANY;
-	/** indicator '*' **/
-	public final static int ZERO_OR_MORE = ZERO | ONE | MANY;
-	/** indicator '?' **/
-	public final static int ZERO_OR_ONE = ZERO | ONE;
-    
-    public final static String toString(int cardinality) {
-        switch(cardinality) {
-            case EMPTY:
+public enum Cardinality {
+
+    EMPTY_SEQUENCE(ZERO),
+
+    //TODO(AR) can we eliminate this?
+    EXACTLY_ONE(ONE),
+
+    //TODO(AR) eliminate this in favour of probably ONE_OR_MORE
+    _MANY(MANY),
```

Comment:
Since `_MANY` is used outside of this class I would switch to the internal value being called `_MANY` and the one used outside to use the better readable one. 
Example for an expression: `Cardinality.MANY...` instead of `Cardinality._MANY`

Reply:
I considered that, but decided against it as `_MANY` is a temporary stop-gap for eXist-db and not something we have in FusionDB. So in an effort to keep the code as similar as possible, I just added `_MANY` for eXist-db.